### ISSUE TITLE
[ENHANCEMENT]: Perf guide

### ISSUE THREAD

**Issue by sleeepyjack** [NVIDIA/cuCollections#250]
### Is your feature request related to a problem? Please describe.

cuCollections exposes a set of knobs that allow optimizing a hashing data structure for a specific use case.

For example:
 - which probing scheme should I use?
 - what's the best CG size?
 - how does the input data type affect performance?
 - can I use particular operations concurrently? How does that impact performance?

The interaction between those choices is also non-trivial.
Finding out which combination works best for an application is a time-consuming task.

### Describe the solution you'd like

Write a perf guide. Could be as simple as a Markdown file.

**Comment by PointKernel** [comment]
We do provide performance guidance in the probing sequence doc, e.g.:

- https://github.com/NVIDIA/cuCollections/blob/4bdf6063de349be7af8da987cea743aa88e28470/include/cuco/probe_sequences.cuh#L26-L28
- https://github.com/NVIDIA/cuCollections/blob/4bdf6063de349be7af8da987cea743aa88e28470/include/cuco/probe_sequences.cuh#L52-L55

Having a performance tuning section in `README` doesn't seem right.

**Comment by sleeepyjack** [comment]
Right. This would be too mich information for a readme. I would put it in a separate file and link to it from the readme.
